resources/assets/coffee/_classes/forum-topic-reply.coffee

```diff
@@ -115,6 +124,49 @@ class @ForumTopicReply
     @setState 'text', @input[0].value
 
 
+  fetchPreview: =>
+    return if $(@previewButton).hasClass('active')
+
+    if $(@previewButton).attr('data-should-refresh') == '0'
+      @showPreview()
+      return
+
+    $.ajax
+      url: $(@previewButton).attr('data-preview-url')
+      method: 'POST'
+      data:
+        body: $(@input).val()
+
+    .done (data) =>
+      $(@previewBox).html(data)
+
+      $(@previewButton).attr('data-should-refresh', '0')
+      $(@input).one 'input', @invalidatePreview
+
+      @showPreview()
+
+
+  showPreview: =>
```

Comment:
both this and the other one need to refresh height of invisible container (for example by calling `osu.pageChange()`)